File: pages/notetaker.page.ts
import { expect, Page } from '@playwright/test';
import { BasePage } from './base.page';

export class NotetakerPage extends BasePage {
  constructor(page: Page) { super(page); }
  async open(): Promise<void> { await this.page.goto('/ai-notes', { waitUntil: 'domcontentloaded' }); }
  async expectReady(): Promise<void> {
    await expect(this.page.getByRole('heading', { name: 'Notes & Recordings' })).toBeVisible();
    await expect(this.page.getByRole('button', { name: 'Ask AI' })).toBeVisible();
  }
}

File: pages/notetaker_page.py
"""Page Object Model for the user-application AI Notetaker (Notes) workspace."""

from playwright.sync_api import expect

from .base_page import BasePage


class NotetakerPage(BasePage):
    """Encapsulates the Notes & Recordings / AI Notetaker workspace."""

    def open(self) -> None:
        """Navigate to the AI Notetaker workspace, waiting for the DOM to be ready."""
        self.page.goto("/ai-notes", wait_until="domcontentloaded")

    def expect_ready(self) -> None:
        """Assert the Notetaker workspace is rendered (heading and Ask AI button)."""
        expect(self.page.get_by_role("heading", name="Notes & Recordings")).to_be_visible()
        expect(self.page.get_by_role("button", name="Ask AI")).to_be_visible()
